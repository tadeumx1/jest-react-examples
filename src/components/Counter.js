import React, { Component } from 'react';

class Counter extends Component {

  render() {
    return (
      <div>
          <h2> VocÃª tem { this.props.count * 2 } counters </h2>
      </div>
    );
  }
}

export default Counter;