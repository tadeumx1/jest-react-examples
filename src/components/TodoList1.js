import React, { Component, Fragment } from 'react';

import { graphql, compose } from 'react-apollo';
import gql from 'graphql-tag';

export class TodoList1 extends Component {

    state = {
        newTodoText: '',
    }

    /* addTodo = async () => {

        const { newTodoText } = this.state;
    
        await this.props.addTodo({
          variables: { text: newTodoText },
          update: (proxy, { data: { createTodo } }) => {
            this.props.todos.refetch();
          },
        })

    }; 

    removeTodo = async (idTodo) => {
    
        await this.props.removeTodo({
          variables: { id: idTodo },
          update: (proxy, { data: { removeTodo } }) => {
            this.props.todos.refetch();
          },
        })

    }; */

    renderTodoList = async () => {

        // console.log(this.props.todos.allTodoes);

        /* <ul>

            { this.props.allTodoes.map(todo => (

                <li key={todo.id}>{todo.text}</li>

            )) }

        </ul> */

    }

  render() {

    // console.log(this.props);

    const { todos } = this.props;

    return (

        <Fragment>

        {
             // this.renderTodoList()
        
            <ul>

            { this.props.allTodoes.map(todo => (

                <li key={todo.id}>{todo.text}
                <input type="submit" value="Remover" onClick={() => this.removeTodo(todo.id)} />
                </li>

            )) }

            </ul>        
        }

        <input
          type="text"
          value={this.state.newTodoText}
          onChange={e => this.setState({ newTodoText: e.target.value })}
        />

        <input type="submit" value="Criar" onClick={() => this.addTodo()} />

        </Fragment>
      
    );
  }
}

const TodosQuery = gql`

    query {

        allTodoes {

            id
            text

        }


    }

`;

const TodoMutation = gql`
  mutation ($text: String!) {
    createTodo ( text: $text ) {
      id
      text
      completed
    }
  }
`;

/*

  mutation ($id: ID!) {
  
        deleteTodo(id: $id) (
      
            id
      
        ) {
    
        id
        text
    
       }
    
    }

*/

const TodoRemoveMutation = gql`
    
mutation removeTodo($id: ID!) {

    deleteTodo(id: $id) {
      id
      text
    }
    
  }
  
`;

// Usando a comoposição do React

/* export default compose(

    graphql(TodoMutation, { name: 'addTodo' }),
    graphql(TodoRemoveMutation, { name: 'removeTodo' }),
    graphql(TodosQuery)

)(TodoList1); */

const TodoList1Wrapper = compose(

    graphql(TodoMutation, { name: 'addTodo' }),
    graphql(TodoRemoveMutation, { name: 'removeTodo' }),
    graphql(TodosQuery, { name: 'todos' })

)(TodoList1);

export default TodoList1Wrapper;
