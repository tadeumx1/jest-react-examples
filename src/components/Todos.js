import React, { Component } from 'react';

import { connect } from 'react-redux';

class Todos extends Component {

  render() {
    return (
      <ul className="Todos">

        { this.props.todos.map(todo => (

            <li key={todo.id}>{todo.text}</li>

        )) }

      </ul>
    );
  }
}

// Utilizando a composição do React

const mapStateToProps = state => ({

    todos: state.todos,

});

export default connect(mapStateToProps)(Todos);
