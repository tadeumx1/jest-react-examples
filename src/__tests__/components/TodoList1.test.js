import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import { ApolloProvider } from 'react-apollo';

import apolloClient from '../../services/apollo';

import { TodoList1 } from '../../components/TodoList1';

describe('Testing Todos Component', () => {
  it('should render correctly', () => {

    const wrapper = shallow(

      <TodoList1 allTodoes={
          [
            {
              "id": "cjlsbtne80tg50182ep34ielv",
              "text": "Fazer cafÃ©"
            },
            {
              "id": "cjlsbtttz0tg90182jhya4xs2",
              "text": "Estudar GraphQL"
            }
          ]
        }/>


    )
    expect(wrapper).toMatchSnapshot();

  });

});
