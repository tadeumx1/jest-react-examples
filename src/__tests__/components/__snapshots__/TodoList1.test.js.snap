// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Todos Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoList1
    allTodoes={
        Array [
            Object {
              "id": "cjlsbtne80tg50182ep34ielv",
              "text": "Fazer café",
            },
            Object {
              "id": "cjlsbtttz0tg90182jhya4xs2",
              "text": "Estudar GraphQL",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ul>
          <li>
                    Fazer café
                    <input
                              onClick={[Function]}
                              type="submit"
                              value="Remover"
                    />
          </li>
          <li>
                    Estudar GraphQL
                    <input
                              onClick={[Function]}
                              type="submit"
                              value="Remover"
                    />
          </li>
</ul>,
        <input
          onChange={[Function]}
          type="text"
          value=""
/>,
        <input
          onClick={[Function]}
          type="submit"
          value="Criar"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              Fazer café
              <input
                            onClick={[Function]}
                            type="submit"
                            value="Remover"
              />
</li>,
            <li>
              Estudar GraphQL
              <input
                            onClick={[Function]}
                            type="submit"
                            value="Remover"
              />
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "cjlsbtne80tg50182ep34ielv",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Fazer café",
                <input
                  onClick={[Function]}
                  type="submit"
                  value="Remover"
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Fazer café",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "onClick": [Function],
                  "type": "submit",
                  "value": "Remover",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "cjlsbtttz0tg90182jhya4xs2",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Estudar GraphQL",
                <input
                  onClick={[Function]}
                  type="submit"
                  value="Remover"
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Estudar GraphQL",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "onClick": [Function],
                  "type": "submit",
                  "value": "Remover",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onClick": [Function],
          "type": "submit",
          "value": "Criar",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ul>
            <li>
                        Fazer café
                        <input
                                    onClick={[Function]}
                                    type="submit"
                                    value="Remover"
                        />
            </li>
            <li>
                        Estudar GraphQL
                        <input
                                    onClick={[Function]}
                                    type="submit"
                                    value="Remover"
                        />
            </li>
</ul>,
          <input
            onChange={[Function]}
            type="text"
            value=""
/>,
          <input
            onClick={[Function]}
            type="submit"
            value="Criar"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                Fazer café
                <input
                                onClick={[Function]}
                                type="submit"
                                value="Remover"
                />
</li>,
              <li>
                Estudar GraphQL
                <input
                                onClick={[Function]}
                                type="submit"
                                value="Remover"
                />
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "cjlsbtne80tg50182ep34ielv",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Fazer café",
                  <input
                    onClick={[Function]}
                    type="submit"
                    value="Remover"
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Fazer café",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "onClick": [Function],
                    "type": "submit",
                    "value": "Remover",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "cjlsbtttz0tg90182jhya4xs2",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Estudar GraphQL",
                  <input
                    onClick={[Function]}
                    type="submit"
                    value="Remover"
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Estudar GraphQL",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "onClick": [Function],
                    "type": "submit",
                    "value": "Remover",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onClick": [Function],
            "type": "submit",
            "value": "Criar",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
